ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -51,12 +51,10 @@ public struct SignalProducer<T, E: ErrorType> {
 	/// then complete, or immediately send an error, depending on the given
 	/// Result.
 	public init(result: Result<T, E>) {
-		switch result {
```

Comment:
Hm. This is a problem I hadn’t foreseen. Maybe `Result` should be just a simple `enum` and depend on Box.framework too?